src/Elliptic/Executables/NonlinearEllipticSolver.hpp

```diff
@@ -0,0 +1,245 @@
+#include "Elliptic/DiscontinuousGalerkin/SubdomainOperator/InitializeSubdomain.hpp"
+#include "Elliptic/DiscontinuousGalerkin/SubdomainOperator/SubdomainOperator.hpp"
+#include "Elliptic/DiscontinuousGalerkin/Tags.hpp"
+#include "Elliptic/Protocols/FirstOrderSystem.hpp"
+#include "Elliptic/SubdomainPreconditioners/MinusLaplacian.hpp"
+#include "Elliptic/Tags.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "NumericalAlgorithms/Convergence/Tags.hpp"
+#include "Options/Options.hpp"
+#include "ParallelAlgorithms/Actions/AddComputeTags.hpp"
+#include "ParallelAlgorithms/LinearSolver/Actions/MakeIdentityIfSkipped.hpp"
+#include "ParallelAlgorithms/LinearSolver/Gmres/Gmres.hpp"
+#include "ParallelAlgorithms/LinearSolver/Multigrid/Actions/RestrictFields.hpp"
+#include "ParallelAlgorithms/LinearSolver/Multigrid/Multigrid.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Actions/CommunicateOverlapFields.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Actions/ResetSubdomainSolver.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Schwarz.hpp"
+#include "ParallelAlgorithms/LinearSolver/Tags.hpp"
+#include "ParallelAlgorithms/NonlinearSolver/NewtonRaphson/NewtonRaphson.hpp"
+#include "ParallelAlgorithms/NonlinearSolver/Tags.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/AnalyticSolution.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/Background.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialGuess.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace elliptic::nonlin_solver {
+
+namespace OptionTags {
+
+struct NonlinearSolverGroup {
+  static std::string name() { return "NonlinearSolver"; }
+  static constexpr Options::String help = "The iterative nonlinear solver";
+};
+struct NewtonRaphsonGroup {
+  static std::string name() { return "NewtonRaphson"; }
+  static constexpr Options::String help =
+      "Options for the Newton-Raphson nonlinear solver";
+  using group = NonlinearSolverGroup;
+};
+struct LinearSolverGroup {
+  static std::string name() { return "LinearSolver"; }
+  static constexpr Options::String help =
+      "The iterative Krylov-subspace linear solver";
+};
+struct GmresGroup {
+  static std::string name() { return "Gmres"; }
+  static constexpr Options::String help = "Options for the GMRES linear solver";
+  using group = LinearSolverGroup;
+};
+struct SchwarzSmootherGroup {
+  static std::string name() { return "SchwarzSmoother"; }
+  static constexpr Options::String help = "Options for the Schwarz smoother";
+  using group = LinearSolverGroup;
+};
+struct MultigridGroup {
+  static std::string name() { return "Multigrid"; }
+  static constexpr Options::String help = "Options for the multigrid";
+  using group = LinearSolverGroup;
+};
+
+}  // namespace OptionTags
+
+/*!
+ * \brief A complete nonlinear elliptic solver stack. Use to compose an
+ * executable.
+ */
+template <typename Metavariables>
+struct Solver {
```

Comment:
Have a protocol or document what this assumes will be in the Metavariables? I find the cyclic `using ... = ...<Metavariables>` leads to extremely cryptic errors because of parse order. It would be better to explicitly take the things you need (seems only system and volume dim)

Reply:
Unfortunately the `Metavariables` template parameter is needed to instantiate parallel components. I agree it would be nicer to disentangle the hierarchy, not really sure how to do that at the moment though. Added some docs for now.

Reply:
Ah right, okay. Thanks!